utils/hints_test.go

```diff
@@ -219,7 +242,14 @@ func TestGenerateHints(t *testing.T) {
 				continue
 			}
 		}
-		assert.Equal(t, test.result, GenerateHints(annMap, "foobar", "co.elastic"))
+		generateHints, incorrectHints := GenerateHints(annMap, "foobar", "co.elastic", allSupportedHints)
+		//Only in Logs_multiline_and_metrics_with_metrics_path we have added co.elastic.hints.steam annotation with a typo error
+		if test.name == "Logs_multiline_and_metrics_with_metrics_path" {
+			assert.Equal(t, 1, len(incorrectHints)) // We validate how many incorrect hints are provided in test1.
```

Comment:
Not a blocker, but this seems really hard coded, maybe add a new field to test struct with the expected length instead of checking it this way

Reply:
Thank you @constanca-m . Done in [cea860a](https://github.com/elastic/elastic-agent-autodiscover/pull/81/commits/cea860a13f1f3608f2ef9619752a6cc6827bf302)